trigger-and-filter/components/deploy-filter-dialog/deploy-filter-dialog.component.ts

```diff
@@ -0,0 +1,128 @@
+
+    context: XoRuntimeContext = this.injectedData.runtimeContext;
+    triggerInstance: string;
+
+    runtimeContextDataWrapper: XcAutocompleteDataWrapper<XoRuntimeContext> = new XcAutocompleteDataWrapper<XoRuntimeContext>(
+        () => this.context,
+        value => {
+            this.context = value;
+            this.fillTriggerInstanceWrapper();
+        });
+
+    triggerInstanceDataWrapper: XcAutocompleteDataWrapper<string> = new XcAutocompleteDataWrapper<string>(
+        () => this.triggerInstance,
+        value => {
+            this.triggerInstance = value;
+        });
+
+    constructor(injector: Injector,
+        private readonly apiService: ApiService,
+        private readonly dialogService: XcDialogService,
+        private readonly i18n: I18nService,
+        private readonly cdr: ChangeDetectorRef) {
+        super(injector);
+
+        this.fillContextWrapper();
+        this.runtimeContextDataWrapper.valuesChange.subscribe({
+            next: () => this.fillTriggerInstanceWrapper()
+        });
+    }
+
+    fillContextWrapper() {
+        this.apiService.startOrderAssertFlat<XoRuntimeContext>(FM_RTC, ORDER_TYPES.POSSIBLE_CONTEXT_FILTER, this.injectedData, XoRuntimeContextArray)
+            .subscribe({
+                next: result => {
+                    this.runtimeContextDataWrapper.values = result.map(rtc => <XcOptionItem<XoRuntimeContext>>{ name: rtc.label, value: rtc });
+                },
+                error: err => {
+                    this.dialogService.error(err);
+                },
+                complete: () => {
+                    this.cdr.markForCheck();
+                }
+            });
+    }
+
+    fillTriggerInstanceWrapper() {
+        this.apiService.startOrderAssertFlat<XoTriggerInstance>(FM_RTC, ORDER_TYPES.POSSIBLE_TRIGGER_INSTANCES, [this.injectedData, this.context], XoTriggerInstanceArray)
+            .subscribe({
+                next: result => {
+                    this.triggerInstanceDataWrapper.values = result.map(triggerInstance => XcOptionItemString(triggerInstance.triggerInstance));
+                },
+                error: err => {
+                    this.dialogService.error(err);
+                },
+                complete: () => {
+                    this.cdr.markForCheck();
+                }
+            });
+    }
+
+    deploy() {
+        const request: XoDeployFilterRequest = new XoDeployFilterRequest();
+        request.filterName = this.injectedData.name;
+        request.filterInstanceName = this.instance;
+        request.runtimeContext = this.context;
+        request.triggerInstanceName = this.triggerInstance;
+        request.configurationParameter = this.parameter;
+        request.documentation = this.documentation;
+        request.optional = this.optional;
+
+        this.apiService.startOrder(FM_RTC, ORDER_TYPES.DEPLOY_FILTER, request, null, StartOrderOptionsBuilder.defaultOptionsWithErrorMessage)
+            .subscribe({
```

Comment:
you could also handle errors on the network layer here (inside `subscribe` or by piping a catch)